src/app.test.js

```diff
@@ -40,16 +41,36 @@ const addTodo = (todoList, text, completed) => {
   }
 };
 
-test("renders empty todo list when starts", () => {
+const logInUser = (userName) => {
+  const userInput = screen.getByPlaceholderText("Please enter username");
+  const setUserButton = screen.getByText("Set User");
+
+  userEvent.type(userInput, userName);
+  userEvent.click(setUserButton);
+};
+test("wont allow to add new todo until user is not set", () => {
+  // act
+  const todoList = screen.getByTestId("todo-list");
+  addTodo(todoList, "Take wife for a walk!");
+
+  // assert
+  expect(
+    queryByDisplayValue(todoList, "Take wife for a walk!")
+  ).not.toBeInTheDocument();
+});
```

Comment:
this is interesting, I think checking that `addTodo` does not result in a particular todo added allows too much room and can create a false positive scenario

I believe the code could be change in a way that this test would still pass but a todo item would still be added to the list e.g. the code amends the user input and prefixes it with a `*` or something similar

in this case it'd be better to check the concrete input and button elements that they are disabled which means it is _not possible_ to use them